modules/tensor_mechanics/src/materials/crystal_plasticity/FiniteStrainUObasedCP.C

```diff
@@ -553,15 +561,13 @@ FiniteStrainUObasedCP::calcJacobian()
     std::vector<RankTwoTensor> dtaudpk2(nss), dfpinvdslip(nss);
     std::vector<Real> dslipdtau;
     dslipdtau.resize(nss);
-    _uo_slip_rates[i]->calcSlipRateDerivative(_qp, _dt, dslipdtau);
+    _uo_slip_rates[i]->calcSlipRateDerivative(_qp, _substep_dt, dslipdtau);
     for (unsigned int j = 0; j < nss; j++)
     {
       dtaudpk2[j] = (*_flow_direction[i])[_qp][j];
       dfpinvdslip[j] = -_fp_old_inv * (*_flow_direction[i])[_qp][j];
+      dfpinvdpk2 += (dfpinvdslip[j] * dslipdtau[j] * _substep_dt).outerProduct(dtaudpk2[j]);
     }
-
-    for (unsigned int j = 0; j < nss; j++)
-      dfpinvdpk2 += (dfpinvdslip[j] * dslipdtau[j] * _dt).outerProduct(dtaudpk2[j]);
```

Comment:
Is this change just to remove the extra loop over the `nss` variable, since each product is computed on the individual `j` entries of the vectors?

Reply:
Exactly :)

Reply:
Does the material based version also need this optimization? :-)

Reply:
Oh nice!